botocore/endpoint.py

```diff
@@ -75,7 +77,26 @@ def convert_to_response_dict(http_response, operation_model):
 
 
 class PreserveAuthSession(Session):
+    """Internal session class used to workaround requests behavior.
+
+    This class is intended to be used only by the Endpoint class.
```

Comment:
I can rename it.  Maybe something like `BotocoreHTTPSession` to disambiguate it from `botocore.session.Session`?